### PR TITLE
Access tokens in openid controller are hashed

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
## Description of change
Access tokens in the openid_application controller will be hashed

## Original issue(s)
https://vajira.max.gov/browse/API-6281

## Tests / Regression Tests

```
 ✓ Valid Token. Patient Scp. Valid ICN. Should return 200
 ✓ Valid Token. With Launch/Patient scp. Patient Scp. Valid ICN. Should return 200
 ✓ Valid Token. Patient Scp. Invalid ICN. Should return !200
 ✓ Valid Token. With Launch/Patient scp. Patient Scp. Invalid ICN. Should return !200
 ✓ Patient Token. Valid. With Launch/Patient scp. Return 200 with appropriate fields
 ✓ SSOI Token. Valid. With Launch scp. With valid vista id. Return 200 with appropriate fields
 ✓ SSOI Token. Valid. With Launch scp. With invalid vista id. Return !200
```

image

### CODE REVIEW

**Review by dbrown-va — CHANGES_REQUESTED** [review]
There is one other session look up modification that is needed in the Veteran Verification module surrounding the mock vs real emis support.

https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/veteran_verification/app/controllers/veteran_verification/application_controller.rb#L18

**Comment by beaugrantham-va** [comment]
Can the hashing implementation be moved into a method?

```
def hash_token(token) ...
```

This will let us change the implementation (e.g., SHA-2 -> SHA-3) without having to update it in so many places, risking missing something.

File: modules/veteran_verification/app/controllers/veteran_verification/application_controller.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def authenticate_token
       # Not supported for Client Credentials tokens
       return false if token.blank? || token.client_credentials_token?
 
-      @session = Session.find(token)
+      @session = Session.find(Digest::SHA256.hexdigest(token.to_s))
```
Comment (dbrown-va) [inline modules/veteran_verification/app/controllers/veteran_verification/application_controller.rb:18]: This should have similar `@session = Session.find(token) if @session.nil?` fallback logic, right?

**Review by dbrown-va — APPROVED** [review]
Looks good.

**Pull request merged**